api/envoy/config/filter/http/fault/v2/fault.proto

```diff
@@ -88,4 +88,22 @@ message HTTPFault {
   //  This is a per-stream limit versus a connection level limit. This means that concurrent streams
   //  will each get an independent limit.
   filter.fault.v2.FaultRateLimit response_rate_limit = 7;
+
+  FaultAbortRuntime abort_runtime = 8;
```

Comment:
I still think you are overcomplicating this to a high degree. Just ignore all of the downstream cluster runtime keys. All you need to do is provide a config override for the hard coded keys defined here:

https://github.com/envoyproxy/envoy/blob/e97e42f046ff54fff213c9d800eade7bc42046e5/source/extensions/filters/http/fault/fault_filter.h#L201-L208

All you need to do is load the final key to use into the fault settings at config load time, whether global or route based. I would ignore the downstream cluster keys for now.